src/api/auth.ts

```diff
@@ -0,0 +1,88 @@
+import { HTTP } from '../core/HTTP';
+import { BASE_URL_API } from "./constants";
+import Store from "../core/Store/Store";
+
+const httpInstance = new HTTP;
+
+export type TUser = {
+  "id": number,
+  "display_name": "string",
+  "avatar": "string",
+} & Omit<TSignUpData, 'password'>;
+
+export type TSignUpData = {
+  "first_name": "string",
+  "second_name": "string",
+  "login": "string",
+  "email": "string",
+  "password": "string",
+  "phone": "string"
+}
+
+export type TLoginData = {
+  "login": "string",
+  "password": "string",
+} | {}
+
+export type TSignUp = {
+  data : TSignUpData,
+  headers: {
+    [index: string]: string
+  }
+}
+
+type TLogin = {
+  data : TLoginData,
+  headers: {
+    [index: string]: string
+  }
+}
+
+const AuthAPI = {
+  signUp: async (data: TSignUp) => {
+    try {
+      const res = await httpInstance.post(`${BASE_URL_API}/api/v2/auth/signup`, data)
+      return {
+        status: res.status,
+        response: typeof res.response === 'object' ? JSON.parse(res.response) : res.response
+      }
+    } catch (error) {
+      return error;
+    }
+  },
+
+  login: async (data: TLogin) => {
+    try {
+      const res = await httpInstance.post(`${BASE_URL_API}/api/v2/auth/signin`, data)
+      return {
+        status: res.status,
+        response: typeof res.response === 'object' ? JSON.parse(res.response) : res.response
+      }
+    } catch (error) {
+      return error;
+    }
+  },
+
+  logout: async () => {
+    try {
+      await httpInstance.post(`${BASE_URL_API}/api/v2/auth/logout`);
+      localStorage.removeItem(Store.STORE_NAME);
+    } catch (error) {
+      return error;
+    }
+  },
+
+  getUserInfo: async () => {
```

Comment:
В этом файле есть тип TUserInfo. Если это тот тип, который приходит с API, то имеет смысл объявить его в сигнатуре


```suggestion

  getUserInfo: async () => Response<TUserInfo> {
```

Но придется тогда написать тип Response например так

```
type Response<T> = {
 status: 200 | 300 | ... | 500
 response: T
}
```